Source/drlg_l3.cpp

```diff
@@ -1433,74 +1377,113 @@ bool Spawn(int x, int y, int *totarea)
 	return false;
 }
 
-bool HivePlaceSetRandom(const BYTE *miniset, int rndper)
+bool PlaceMiniSet(const Miniset &miniset, bool setview)
 {
-	bool placed = false;
-	int sw = miniset[0];
-	int sh = miniset[1];
+	int sw = miniset.size.width;
+	int sh = miniset.size.height;
+	int sx = GenerateRnd(DMAXX - sw) - 1;
+	int sy = GenerateRnd(DMAXY - sh);
+
+	for (int bailcnt = 0;; bailcnt++) {
+		if (bailcnt > 198)
+			return false;

```

Comment:
This is why cave level generation is so bad? It only checks ~6 rows to see if it can place the stairs before bailing 😭

Reply:
It at least explains why a large portion of seeds gets skipped, I wouldn't say it's necessarily bad, but it could cause the same levels to appear multiple times i suppose